controllers/articles.controllers.js

```diff
@@ -3,11 +3,19 @@ const {
   selectArticle,
   updateArticle,
 } = require("../models/articles.models");
+const { selectTopics } = require("../models/topics.models");
 
 exports.getAllArticles = (req, res, next) => {
-  selectArticle()
-    .then((articles) => {
-      res.status(200).send({ articles });
+  const { topic } = req.query;
+  const promises = [selectArticle(topic)];
+
+  if (topic) {
+    promises.push(selectTopics(topic));
+  }
```

Comment:
Great use of another db.query to check if the topic exists in the db before searching for it